include/nbl/builtin/hlsl/math/functions.hlsl

```diff
@@ -122,11 +122,7 @@ void frisvad(NBL_CONST_REF_ARG(T) normal, NBL_REF_ARG(T) tangent, NBL_REF_ARG(T)
 
 bool partitionRandVariable(float leftProb, NBL_REF_ARG(float) xi, NBL_REF_ARG(float) rcpChoiceProb)
 {
-#ifdef __HLSL_VERSION
-    NBL_CONSTEXPR float NEXT_ULP_AFTER_UNITY = asfloat(0x3f800001u);
-#else
-    NBL_CONSTEXPR float32_t NEXT_ULP_AFTER_UNITY = bit_cast<float32_t>(0x3f800001u);
-#endif
+    const float32_t NEXT_ULP_AFTER_UNITY = bit_cast<float32_t>(0x3f800001u);
```

Comment:
should be templated on `T` instead of `float` and live in `sampling.hlsl` (named same as the folder and namespace), do as part of #811 

P.S. for the constant you can use
```cpp
const T NextULPAfterUnity = bit_cast<T>(++bit_cast<unsigned_int_of_size_t<sizeof(T)> >(T(1)));
```

Reply:
or you can put it in `sampling/basic.hlsl`